### PR TITLE
Coverage genus

### PR DESCRIPTION
**Pull request by simonbrandhorst**

### CODE REVIEW

File: test/QuadForm/Quad/Genus.jl [hunk 1]
```diff
@@ -45,6 +69,19 @@
...
+    G1 = rand(G)
+    G2 = rand(G)
+    L1 = representative(G1)
+    @test L1 in G1
+    L2 = representative(G2)
+    G3 = @inferred orthogonal_sum(G1, G2)
+    L3, = orthogonal_sum(L1, L2)
+    @test G3 == genus(L3, p2)
+  end
+  print(G)
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:83]:
```suggestion
```

File: test/QuadForm/Quad/Genus.jl [hunk 2]
```diff
@@ -77,6 +119,7 @@
     L3, = orthogonal_sum(L1, L2)
     @test L3 in G3
   end
+  print(G)
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:122]:
```suggestion
```

File: src/QuadForm/Quad/Genus.jl [hunk 3]
```diff
@@ -174,6 +174,7 @@ function JorDec(L::QuadLat, p)
   return JorDec(J, G, E, p)
 end
 
+
```
Comment (thofma) [inline src/QuadForm/Quad/Genus.jl:177]:
```suggestion
```

File: test/QuadForm/Quad/Genus.jl [hunk 4]
```diff
@@ -87,6 +130,44 @@
...
+  end
+  sig = Dict([(pl[1],3)])
+  for d in 1:(long_test ? 10 : 3)
+    for G in Hecke.genera_quadratic(F,rank=3,det=d*OF,signatures=sig)
+      @test representative(G) in G
+    end
+  end
+
+
+
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:170]:
```suggestion
```

File: test/QuadForm/Quad/Genus.jl [hunk 5]
```diff
@@ -87,6 +130,44 @@
   p = prime_decomposition(maximal_order(K), 5)[1][1]
   fl, LL = Hecke.ismaximal_integral(L, p)
   @test !fl
+
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:133]:
```suggestion
```

File: test/QuadForm/Quad/Genus.jl [hunk 6]
```diff
@@ -33,8 +33,34 @@
...
+    L1 = representative(G1)
+    @test L1 in G1
+    L2 = representative(G2)
+    G3 = @inferred orthogonal_sum(G1, G2)
+    L3, = orthogonal_sum(L1, L2)
+    @test G3 == genus(L3, p2)
+    # @test genus(L1,p2) + genus(QuadLat,p2) == G1 broken until 0x0 pseudo matrices have a base_ring
+  end
+  @test sprint(show, G[1]) isa String
+  @test sprint(show, "text/plain", G[1]) isa String)
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:61]:
```suggestion
  @test sprint(show, "text/plain", G[1]) isa String
```

File: test/QuadForm/Quad/Genus.jl [hunk 7]
```diff
@@ -54,7 +92,15 @@
...
     @test G3 == genus(L3, p3)
+    @test genus(L1,p3) + genus(QuadLat,p3) == G1
   end
+  q = genus(QuadLat,p2)
+  @test det(q) == 1
+  @test rank(q) == 0
+  @test hasse_invariant(q) == 1
+  
+  @test sprint(show, G[1]) isa String
+  @test sprint(show, "text/plain", G[1]) isa String)
```
Comment (thofma) [inline test/QuadForm/Quad/Genus.jl:103]:
```suggestion
  @test sprint(show, "text/plain", G[1]) isa String
```

**Comment by simonbrandhorst** [comment]
The tests seem to run very long?

**Comment by thofma** [comment]
I think it was a CI problem. We still need an example with a nonset Jordan decomposition and the zero rank case.

**Comment by simonbrandhorst** [comment]
Bots will fail now in
`_quadratic_unimodular_lattice_dyadic(p, r, w, d, alpha, wi)`.
There is some bug.

**Comment by thofma** [comment]
Nooooo, we did not have 100% because we missed the nondyadic case.

**Comment by simonbrandhorst** [comment]
Haha. Well do a new PR, then :D.

**Pull request merged**